cmd/federation-controller/main.go

```diff
@@ -134,21 +135,20 @@ func main() {
 	}()
 
 	var fdsClient *adsc.ADSC
-
 	remote := cfg.MeshPeers.Remote
-
 	if len(remote.Addresses) > 0 {
-		remoteName := remote.Name
 		var discoveryAddr string
-		if remote.IngressType == config.OpenShiftRouter {
-			discoveryAddr = remote.Addresses[0]
+		if networking.IsIP(remote.Addresses[0]) {
+			discoveryAddr = fmt.Sprintf("federation-discovery-service-%s.istio-system.svc.cluster.local:15080", remote.Name)
```

Comment:
I wonder if we shouldn't encapsulate it in remote struct. The naming is something to polish further, but I was thinking about:

```golang
hostname, port := remote.Service()
discoveryAddr := fmt.Sprintf("federation-discovery-service-%s.istio-system.svc.cluster.local:%d", hostname, port)
```
where hostname can be also used as authority.

And impl:

```golang
func (r *Remote) Service() (string, uint32) {
	hostname := r.Addresses[0]
	if networking.IsIP(hostname) {
		hostname = fmt.Sprintf("federation-discovery-service-%s.istio-system.svc.cluster.local", r.Name)
	}

	return hostname, 15080
}
```

This would also limit duplication in all `config_factory.go`

Reply:
Good idea, but I splitted functions for port, service name and service fqdn. Please look if that is ok?